dh_fixperms

```diff
@@ -91,76 +118,49 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 	}
 
 	# General permissions fixing.
-	complex_doit("find $tmp $find_options -print0",
+	complex_doit("find $tmp ${find_exclude_options} -print0",
 		"2>/dev/null | xargs -0r chown --no-dereference 0:0");
-	complex_doit("find $tmp ! -type l $find_options -print0",
-		"2>/dev/null | xargs -0r chmod go=rX,u+rw,a-s");
+	find_and_reset_perm($tmp, 'go=rX,u+rw,a-s', '! -type l');
 	
 	# Fix up permissions in usr/share/doc, setting everything to not
 	# executable by default, but leave examples directories alone.
-	complex_doit("find $tmp/$prefix/share/doc -type f $find_options ! -regex '$tmp/usr/share/doc/[^/]*/examples/.*' -print0 2>/dev/null",
-		"| xargs -0r chmod 0644");
-	complex_doit("find $tmp/$prefix/share/doc -type d $find_options -print0 2>/dev/null",
-		"| xargs -0r chmod 0755");
-
-	# Executable man pages are a bad thing..
-	complex_doit("find $tmp/$prefix/share/man -type f",
-		"$find_options -print0 2>/dev/null | xargs -0r chmod 0644");
-
-	# ..and header files ..
-	complex_doit("find $tmp/$prefix/include -type f $find_options -print0",
-		"2>/dev/null | xargs -0r chmod 0644");
-	
-	# ..and desktop files ..
-	complex_doit("find $tmp/$prefix/share/applications -type f $find_options -print0",
-		"2>/dev/null | xargs -0r chmod 0644");
-	
-	# .. and perl modules.
-	complex_doit("find $tmp/$perlvendorarch $tmp/$perlvendorlib -type f",
-		"-perm -5 -name '*.pm' $find_options -print0",
-		"2>/dev/null | xargs -0r chmod a-X");
-
-	complex_doit("find $tmp -perm -5 -type f ${mode_0644_find_pattern}",
-				 "${find_options} -print0 2>/dev/null",
-				 "| xargs -0r chmod 0644");
+	find_and_reset_perm("${tmp}/${prefix}/share/doc", '0644', '-type f', "! -regex '$tmp/${prefix}/share/doc/[^/]*/examples/.*'");
+	find_and_reset_perm("${tmp}/${prefix}/share/doc", '0755', '-type d');
+
+	# Manpages, include file, desktop files, etc., shouldn't be executable
+	find_and_reset_perm([
+				"${tmp}/${prefix}/share/man",
+				"${tmp}/${prefix}/include",
+				"${tmp}/${prefix}/share/applications",
+				"${tmp}/${prefix}/share/lintian/overrides",
+			], '0644', '-type f');
+
+	# nor should perl modules.
+	find_and_reset_perm(["${tmp}/${perlvendorarch}", "${tmp}/${perlvendorlib}"],
+			'a-X', "-type f -perm -5 -name '*.pm'");
+
+	find_and_reset_perm($tmp, '0644', '-type f ' . patterns2find_expr(@mode_0644_patterns)) if @mode_0644_patterns;
+	find_and_reset_perm($tmp, '0755', '-type f ' . patterns2find_expr(@mode_0755_patterns)) if @mode_0755_patterns;
 	
 	# Programs in the bin and init.d dirs should be executable..
-	for my $dir ("$prefix/bin", "bin", "$prefix/sbin", "sbin", "$prefix/games", "etc/init.d") {
-		if (-d "$tmp/$dir") {
-			complex_doit("find $tmp/$dir -type f $find_options -print0 2>/dev/null",
-				"| xargs -0r chmod a+x");
-		}
-	}
+	find_and_reset_perm([map { "${tmp}/$_"} @executable_files_dirs], 'a+x', '-type f');
```

Comment:
This isn't going to work quite right. `@executable_files_dirs` is `usr/bin` and friends, but `usr/` needs to be replaced with `$prefix/`. I'm not sure why that was split out to a toplevel array, anyways. It's no different than the "shouldn't be executable" call above.

Although this changes the original array, I think you can change to this:
```
map { s,^usr/,${prefix}/,; "${tmp}/$_" } @executable_files_dirs
```